src/main/java/dev/drugowick/threehundredsixty/validation/NewPasswordFieldsMustMatch.java

```diff
@@ -0,0 +1,24 @@
+package dev.drugowick.threehundredsixty.validation;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({ TYPE })
+@Retention(RUNTIME)
+@Constraint(validatedBy = { NewPasswordFieldsMustMatchValidator.class })
+public @interface NewPasswordFieldsMustMatch {
```

Comment:
Conforme eu comentei no PR #30 eu fiz não usando Anotations, ainda não dominei bem, então preferi fazer na mão mesmo... Mas fica muito legal isso!

Reply:
Saquei. Aqui a questão é que a gente meio que não tem uma camada de serviços. A gente vai do controller pros repositórios e num cenário ideal teríamos uma camada de serviço com regras de negócio sendo validadas.

Nesse caso o erro que tá dando é na persistência. É provável que seja um tratamento que não faça sentido fazer no Controller mas que não tá sendo feito numa camada intermediária de Serviço, que aí sim trataria exceções de persistência... devolvendo pra camada de Controllers algo já mais traduzido, como uma exceção de negócio, não essa coisa feia de transação "vazando" pra web.

Mas tem formas de mitigar isso aí. Esse erro foi onde? Por exemplo, nesse caso, se é a entidade Employee, ela tá sendo validada com @Valid? Se não é o caso por ela não ser uma entidade que vem do cliente, tem como chamar programaticamente a validação antes de mandar pro repositório persistir.

Se o erro não é de validade da entidade, ou seja, se tem alguma outra regra de negócio além da "corretude" da entidade, aí é falta, mesmo, de uma camada de serviço fazendo a validação e traduzindo as exceções de persistência em exceções de negócio (por exemplo, uma EntidadeDuplicadaException.java).

Se a questão for validar o que tá vindo do cliente, mesmo, pra ir além do @Valid, tem os Mixin do Jackson, que são pra você ter a entidade lá na sua camada de  domínio e outra classe definindo anotações específicas pra outra camada, como a web, separando melhor as responsabilidades. Em último caso ainda tem os DTOs (eu usei um pra uma das validações, enquanto pra outra não).

Reply:
Olha como nesse caso não "estourou" nada (vide console ao fundo) e o usuário recebeu certinho a validação na tela:
![image](https://user-images.githubusercontent.com/7481200/80056895-afd74580-84fb-11ea-88a7-cbdd4cce4200.png)

O @Valid na hora de injetar no controller esse objeto vindo do form se encarregou de tratar as validações e colocá-las no BindingResult pra mim. Eu me encarreguei de verificar se tinha erros e retornar essa info pro cliente.

Reply:
A vantagem de fazer  a anotação custom lá é que mais tarde se a gente for estender a solução pra ter uma API a gente pode validar os DTOs (ou Command Objects, enfim, cada um chama do que quiser) de entrada da API com exatamente a mesma anotação, sem mudar nada.

Reply:
Comentário aleatório: pfv põem `` no **`@Valid`**, caso contrário vai ficar notificando o user Jon Hughes, à toa :laughing: 

Então @brunodrugowick , não lembro exatamente onde foi stackstrace, salvei para olhar depois, mas não lembro o passo-a-passo. :sweat_smile: 

Preciso estudar melhor as coisas que você usou, mas já manda pra dentro o merge! :muscle:

Reply:
Kkkkkkkkkkkkkkkkk. Coitado do cara!